### PR TITLE
ENG-16577 add retries to remove action blocker

### PR DESCRIPTION
**Pull request by zeemanhe**
Action blockers are not removed cleanly under some circumstances. In the presence of  some action blockers, further actions, such as UAC or rejoining,  may be blocked.   Add a retry mechanism to ensure their removal.

### CODE REVIEW

File: src/frontend/org/voltdb/iv2/MpRepairTask.java [hunk 1]
```diff
@@ -82,6 +81,10 @@ public void run(SiteProcedureConnection connection) {
                     try {
                         algo.start().get();
                         repairLogger.info(whoami + "finished repair.");
+                        if (!m_leaderMigration) {
```
Comment (luyangco) [inline src/frontend/org/voltdb/iv2/MpRepairTask.java]: Why here doesn't need to wait for leader promotion completion?

File: src/frontend/org/voltdb/iv2/MpRepairTask.java [hunk 2]
```diff
@@ -82,6 +81,10 @@ public void run(SiteProcedureConnection connection) {
...
+                            // At this point, all sp repairs are completed. Remove the mp repair blocker
+                            VoltZK.removeActionBlocker(m_mailbox.m_messenger.getZK(), VoltZK.mpRepairInProgress, repairLogger);
```
Comment (luyangco) [inline src/frontend/org/voltdb/iv2/MpRepairTask.java]: Is m_mailbox.m_messenger always not NULL?

File: src/frontend/org/voltdb/VoltZK.java [hunk 3]
```diff
@@ -555,6 +557,40 @@ public static boolean removeActionBlocker(ZooKeeper zk, String node, VoltLogger
...
+        while (failedCount < maxCount) {
+          try {
+            zk.delete(path, -1);
+            return true;
+          } catch (KeeperException e) {
+              switch (e.code()) {
+              case OPERATIONTIMEOUT:
+                  failedCount++;
+                  try {
+                      Thread.sleep(100 * failedCount);
```
Comment (luyangco) [inline src/frontend/org/voltdb/VoltZK.java]: Unless there are direct evidences of zookeeper throws the OPERATION_TIMEOUT type of exception, I don't feel comfortable to have this in the transactional path. Also retrying 5 times doesn't guarantee the blocker node being deleted successfully.

File: src/frontend/org/voltcore/zk/ZooKeeperLock.java [hunk 4]
```diff
@@ -113,9 +113,12 @@ public void processResult(int rc, String path, Object ctx) {
...
         if (m_currentPath != null) {
             try {
-                m_zk.delete(m_currentPath, -1);
+                if (!VoltZK.deleteZNodeWithRetries(m_zk, m_currentPath)) {
+                    hostLog.warn("Failed to delete ZK lock:" + m_currentPath);
+                }
                 m_currentPath = null;
             } catch (KeeperException | InterruptedException e) {
+                hostLog.warn("Failed to delete ZK lock:" + m_currentPath + " " + e.getMessage());
                 throw new IOException(e);
```
Comment (luyangco) [inline src/frontend/org/voltcore/zk/ZooKeeperLock.java]: Have we seen the exception being thrown in the test log files?

File: src/frontend/org/voltdb/iv2/MpRepairTask.java (hunk 2 above)
Comment (zeemanhe) [reply src/frontend/org/voltdb/iv2/MpRepairTask.java]: should be not NULL

File: src/frontend/org/voltdb/iv2/MpRepairTask.java (hunk 1 above)
Comment (zeemanhe) [reply src/frontend/org/voltdb/iv2/MpRepairTask.java]: SP promotions should have be completed before MP repair is done.

File: src/frontend/org/voltdb/VoltZK.java (hunk 3 above)
Comment (zeemanhe) [reply src/frontend/org/voltdb/VoltZK.java]: The delete transaction could be interrupted.  How about removing action blocker async?

**Pull request closed without merging**